src/main/java/core/basesyntax/io/read/impl/ReportReaderImpl.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.io.read.impl;
+
+import core.basesyntax.io.read.ReportReader;
+import java.io.FileNotFoundException;
+
+public class ReportReaderImpl implements ReportReader {
+    private final ReportReader reader;
+
+    public ReportReaderImpl(ReportReader reader) {
+        this.reader = reader;
+    }
+
+    public String readReport() throws FileNotFoundException {
```

Comment:
According to the checklist, we should avoid using plain `throws` for exceptions. Instead, use a try-catch block and rethrow a RuntimeException with an informative message if necessary. Also, consider using try-with-resources if you are reading from a file.